src/components/PermissionOption/index.tsx

```diff
@@ -67,7 +70,17 @@ const PermissionOption: React.FC<PermissionOptionProps> = ({
               hasPermission(requirement.permissions, currentPermission, {
                 type: requirement.type ?? 'and',
               })
-            ))
+            )) ||
+          ((option.permission === Permission.REQUEST_4K ||
+            option.permission === Permission.AUTO_APPROVE_4K) &&
+            (!settings.currentSettings.movie4kEnabled ||
+              !settings.currentSettings.series4kEnabled)) ||
+          ((option.permission === Permission.REQUEST_4K_MOVIE ||
+            option.permission === Permission.AUTO_APPROVE_4K_MOVIE) &&
+            !settings.currentSettings.movie4kEnabled) ||
+          ((option.permission === Permission.REQUEST_4K_TV ||
+            option.permission === Permission.AUTO_APPROVE_4K_TV) &&
+            !settings.currentSettings.series4kEnabled)
```

Comment:
This is very difficult to understand. I think it's time we break out these checks from being inline. Instead of repeating ourselves constantly in this file.

Reply:
Done, and added comments to clarify each check 😺